src/contracts/atlas/GasAccounting.sol

```diff
@@ -154,37 +186,31 @@ abstract contract GasAccounting is SafetyLocks {
         uint256 _claims = claims;
         uint256 _withdrawals = withdrawals;
         uint256 _deposits = deposits;
-        uint256 _supply = totalSupply;
-
-        EscrowAccountBalance memory solverEscrow = _balanceOf[winningSolver];
-        uint256 _solverBonded = solverEscrow.bonded;
 
-        // Remove any remaining gas from the bundler's claim.
+        // Remove any unused gas from the bundler's claim.
+        // TODO: consider penalizing bundler for too much unused gas (to prevent high escrow requirements for solvers)
         uint256 gasRemainder = (gasleft() * tx.gasprice);
         gasRemainder += ((gasRemainder * SURCHARGE) / SURCHARGE_BASE);
         _claims -= gasRemainder;
 
-        if (_deposits < _claims + _withdrawals) revert InsufficientTotalBalance((_claims + _withdrawals) - _deposits);
-
-        uint256 surplus = _deposits - _claims - _withdrawals;
-
-        // Remove the hold
-        // NOTE that holds can also be applied from withdrawals, so make sure we don't remove any non-transitory holds
-        if (surplus > _solverBonded) {
-            surplus -= _solverBonded;
-            solverEscrow.bonded = 0;
-            solverEscrow.total += uint128(surplus);
+        if (_deposits < _claims + _withdrawals) {
+            // CASE: in deficit, subtract from bonded balance
+            uint256 amountOwed = _claims + _withdrawals - _deposits;
+            if (_assign(winningSolver, amountOwed)) revert InsufficientTotalBalance((_claims + _withdrawals) - deposits);
+            _deposits += amountOwed;
```

Comment:
Same as the other comment - deposits, claims, and withdrawals all closed off to solver's balance via _credit() or _assign(), so no need to do this when the _deposits value is no persisted anywhere at end of the function, right?
```suggestion
```